TODO

## From Wenbin
1. Development data mining for gene expression and splicing in different tissues
2. Diversity RNA-Seq only collect single tissue (e.g. seedling) of different lines, could add different species in Zea as well, do a public RNA-Seq pan-genome analysis only using seedling from different lines
3. sQTL in different RIL and diversity population and compared to eQTL results, such as [this](http://science.sciencemag.org/content/352/6285/600).
4. burden test of rare allele using the splicing data, does it still has the [smile pattern](http://www.sciencedirect.com/science/article/pii/S0002929716000045)?

## From Jinliang
1. count RNA-seq reads using [kallisto](https://pachterlab.github.io/kallisto/) for three diversity populations (http://zeabigdata.org/2016/11/RNA-seq-diversity/). 
2. eQTL/GWAS/epistasis studies across multiple tissues with above data.
3. Phenotypic GWAS for the above diversity panels.
4. Incorporating expression data for GWAS.
5. VCAP by different expression categories, i.e. expression levels, tissues, splicing, chromatin status, etc.


## Delivery
1. Counts on different version of maize genomes (v3 and v4).
2. Shiny app or d3 data visulization tools for easy access and data query.

Jinliang Yang
