---
title: "Visualize GWAS results"
author: "Jinliang Yang"
date: "07-19-2019"
output:
  html_document:
    df_print: paged
  html_notebook: default
  pdf_document: default
  word_document: default
---

I created a shiny app for your data. Please run the attached file.
 

 
```{r setup}
# , include=TRUE, warning=FALSE, echo=TRUE, error=FALSE
knitr::opts_knit$set(root.dir=normalizePath('../../'))
knitr::opts_chunk$set(warning=FALSE, message=FALSE, error=FALSE, echo=TRUE)
```


```{r}
library(shiny)
library(ggplot2)
library(plotly)

dat = read.csv("data/subset_assoc.csv",stringsAsFactors=FALSE)

ui <- fluidPage(
    titlePanel("Manhattan Plot for GWAS Results"),
    plotlyOutput("manhattan", height="500px"),
    column(1),
    column(9,sliderInput("threshold", label = "Change the threshold to:",
                min = 5, max = 10, value = 9, step = 0.1, width = "100%"))
)

server <- function(input, output) {

    output$manhattan <- renderPlotly({
        p = ggplot(data=dat, aes(x=chr, y=log10p, color=log10p)) + 
            geom_jitter(width=0.45) +
            geom_hline(yintercept = input$threshold) +
            geom_text(aes(x=max(dat$chr)-1, y=input$threshold+0.2,
                          label=paste("Number of obs. above the threshold:",
                                      sum(dat$log10p>=input$threshold))), color=1) +
            scale_x_continuous(breaks=sort(unique(dat$chr)), minor_breaks=NULL) +
            xlab("Chromosome") + ylab("-log10(P-value)") +
            theme_minimal() + theme(legend.position = "none") +
            scale_colour_gradient(low = "steelblue", high = "red")
        pl = ggplotly(p, tooltip = c("x","y"))
        pl$x$data[[2]]$hoverinfo = 'none'
        pl$x$data[[3]]$hoverinfo = 'none'
        pl
    })
}

shinyApp(ui = ui, server = server)

```


I also find this page online (see part 4): https://www.r-graph-gallery.com/wp-content/uploads/2018/02/Manhattan_plot_in_R.html
 



## GWAS results shared by Gen:

`/common/jyanglab/shared/Gen_Xu/282_GWAS_37_trait/output`


### simple test

```{r, eval=FALSE}
# Load the library
library(qqman)
gwasResults <- read.csv("data/subset_assoc.csv",stringsAsFactors=FALSE)
# Make the Manhattan plot on the gwasResults dataset
manhattan(gwasResults, chr="chr", bp="ps", snp="rs", p="p_score" )
```


```{r}
library(plotly)

# Prepare the dataset
don <- gwasResults %>% 
  
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(BP)) %>% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(gwasResults, ., by=c("CHR"="CHR")) %>%
  
  # Add a cumulative position of each SNP
  arrange(CHR, BP) %>%
  mutate( BPcum=BP+tot) %>%

  # Add highlight and annotation information
  mutate( is_highlight=ifelse(SNP %in% snpsOfInterest, "yes", "no")) %>%

  # Filter SNP to make the plot lighter
  filter(-log10(P)>0.5)
  
# Prepare X axis
axisdf <- don %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

# Prepare text description for each SNP:
don$text <- paste("SNP: ", don$SNP, "\nPosition: ", don$BP, "\nChromosome: ", don$CHR, "\nLOD score:", -log10(don$P) %>% round(2), "\nWhat else do you wanna know", sep="")

# Make the plot
p <- ggplot(don, aes(x=BPcum, y=-log10(P), text=text)) +
    
    # Show all points
    geom_point( aes(color=as.factor(CHR)), alpha=0.8, size=1.3) +
    scale_color_manual(values = rep(c("grey", "skyblue"), 22 )) +
    
    # custom X axis:
    scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
    scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
    ylim(0,9) +

    # Add highlighted points
    geom_point(data=subset(don, is_highlight=="yes"), color="orange", size=2) +
  
    # Custom the theme:
    theme_bw() +
    theme( 
      legend.position="none",
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank()
    )
ggplotly(p, tooltip="text")

```



