---
title: "Visualize GWAS results"
author: "Jinliang Yang"
date: "07-19-2019"
output:
  html_document:
    df_print: paged
  html_notebook: default
  pdf_document: default
  word_document: default
---


```{r setup}
# , include=TRUE, warning=FALSE, echo=TRUE, error=FALSE
knitr::opts_knit$set(root.dir=normalizePath('../../'))
knitr::opts_chunk$set(warning=FALSE, message=FALSE, error=FALSE, echo=TRUE)
```


I also find this page online (see part 4): https://www.r-graph-gallery.com/wp-content/uploads/2018/02/Manhattan_plot_in_R.html
 

### simple test

```{r, eval=FALSE}
# Load the library
install.packages("qqman")
library(qqman)
pval <- read.csv("data/subset_assoc.csv",stringsAsFactors=FALSE)
# Make the Manhattan plot on the gwasResults dataset
manhattan(pval, chr="chr", bp="ps", snp="rs", p="p_score", col = c("black", "gray60") )
```

## GWAS results on `crane`:

`/common/jyanglab/shared/Gen_Xu/282_GWAS_37_trait/output`

```{r}
# Load the library
#install.packages("qqman")
library(qqman)
library(data.table)
pval <- fread("/common/jyanglab/shared/Gen_Xu/282_GWAS_37_trait/output/X20KernelWeight.assoc.txt", data.table=FALSE)

dim(pval)
pval$log10p <- -log10(pval$p_score)
pval <- subset(pval, log10p > 3)
# Make the Manhattan plot on the gwasResults dataset

pdf("graphs/mht-plot.pdf", width = 10, height=3)
manhattan(pval, chr="chr", bp="ps", snp="rs", p="p_score", col = c("black", "gray60") )
dev.off()
```

-----------------------------

### High light SNPs of interest

```{r, eval=FALSE}
# Load the library

chr8 <- subset(pval, chr ==8 & log10p > 7)

# Make the Manhattan plot on the gwasResults dataset
manhattan(pval, chr="chr", bp="ps", snp="rs", p="p_score", col = c("black", "gray60"), highlight = chr8$rs )
```



```{r, eval=FALSE}
library(plotly)

# Prepare the dataset
don <- gwasResults %>% 
  
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(BP)) %>% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(gwasResults, ., by=c("CHR"="CHR")) %>%
  
  # Add a cumulative position of each SNP
  arrange(CHR, BP) %>%
  mutate( BPcum=BP+tot) %>%

  # Add highlight and annotation information
  mutate( is_highlight=ifelse(SNP %in% snpsOfInterest, "yes", "no")) %>%

  # Filter SNP to make the plot lighter
  filter(-log10(P)>0.5)
  
# Prepare X axis
axisdf <- don %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

# Prepare text description for each SNP:
don$text <- paste("SNP: ", don$SNP, "\nPosition: ", don$BP, "\nChromosome: ", don$CHR, "\nLOD score:", -log10(don$P) %>% round(2), "\nWhat else do you wanna know", sep="")

# Make the plot
p <- ggplot(don, aes(x=BPcum, y=-log10(P), text=text)) +
    
    # Show all points
    geom_point( aes(color=as.factor(CHR)), alpha=0.8, size=1.3) +
    scale_color_manual(values = rep(c("grey", "skyblue"), 22 )) +
    
    # custom X axis:
    scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
    scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
    ylim(0,9) +

    # Add highlighted points
    geom_point(data=subset(don, is_highlight=="yes"), color="orange", size=2) +
  
    # Custom the theme:
    theme_bw() +
    theme( 
      legend.position="none",
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank()
    )
ggplotly(p, tooltip="text")

```



